src/lib/optimizer/strategy/chunk_pruning_rule.cpp

```diff
@@ -0,0 +1,100 @@
+#include "chunk_pruning_rule.hpp"
+
+#include <boost/variant.hpp>
+
+#include <algorithm>
+#include <iostream>
+
+#include "all_parameter_variant.hpp"
+#include "constant_mappings.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "optimizer/chunk_statistics.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+std::string ChunkPruningRule::name() const { return "Chunk Pruning Rule"; }
+
+bool ChunkPruningRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  if (node->type() == LQPNodeType::Predicate && node->child_count() == 1) {
+    // try to find a chain of predicate nodes that ends in a leaf
+    std::vector<std::shared_ptr<PredicateNode>> predicate_nodes;
+
+    // Gather adjacent PredicateNodes
+    auto current_node = node;
+    while (current_node->type() == LQPNodeType::Predicate) {
+      predicate_nodes.emplace_back(std::dynamic_pointer_cast<PredicateNode>(current_node));
+      current_node = current_node->left_child();
+      // Once a node has multiple parents, we're not talking about a Predicate chain anymore
+      if (current_node->type() == LQPNodeType::Predicate && current_node->parents().size() > 1) {
+        _apply_to_children(node);
+        return false;
+      }
+    }
+
+    // skip over validation nodes
+    if (current_node->type() == LQPNodeType::Validate) {
+      current_node = current_node->left_child();
+    }
+
+    if (current_node->type() != LQPNodeType::StoredTable) {
+      _apply_to_children(node);
+      return false;
+    }
+    auto stored_table = std::dynamic_pointer_cast<StoredTableNode>(current_node);
```

Comment:
Don't use a `dynamic_pointer_cast` if you don't check for `nullptr`. Static is sufficient here.